observability/microservices.md

```diff
@@ -0,0 +1,53 @@
+
+This is a common issue. When calling other microservices, depending on the technology stack we use, we can hide errors and exceptions that might happen on the other side even without wanting it. If we are using a simple REST interface, the other microservice can return to us a 500 HTTP status code and we don't have any idea what happen inside that microservice.
+
+More important, we don't have any way to associate our Correlation Id to whatever happens inside that microservice. Therefore, is so important to have a plan in place to be able to extend your traceability and monitoring efforts, especially when using a microservice architecture.
+
+## How to extend your tracing information between microservices
+
+The W3C consortium is working in a [Trace Context](https://www.w3.org/TR/trace-context/) definition that we can use when using HTTP as the protocol in our microservice architecture. But let's explain how we can implement this functionality in our software.
+
+The main idea behind this is to propagate the correlation information between http request so other pieces of software can read this information and correctly correlate telemetry across microservices.
+
+The way to propagate this information is to use HTTP Headers for the Correlation Id, parent Correlation Id, etc.
+
+When you are in the scope of a HTTP Request, your tracing system should already have created four properties that you can use to send across your microservices.
+
+- RequestId:0HLQV2BC3VP2T:00000001,
+- SpanId:da13aa3c6fd9c146,
+- TraceId:f11a03e3f078414fa7c0a0ce568c8b5c,
+- ParentId:5076c17d0a604244
+
+This is an example of the four properties you can find which identify the current request.
+
+- RequestId is the unique id that represent the current HTTP Request.
+- SpanId is the default automatically generated span. You can have more than one Span that scope different functionality inside your software.
+- TraceId represent the id for current log trace.
+- ParentId is the parent span id, that in some case can be the same or something different.
+
+## Example
+
+Now we are going to explore an example with 3 microservices that calls to each other in a row.
+
+![image](./microservices.png)
+
+This image is the summary of what is needed in each microservice to propagate the trace-id from A to C.
+
+The root caller is A and that is why it doesn't have a parent-id, only have a new trace-id. Next, A calls B using HTTP. To propagate the correlation information as part of the request, we are using two new headers, that based on the W3C Correlation specification, are: trace-id and parent-id. So, in this example because A is the root caller, he only sends his own trace-id to the microservice B.
```

Comment:
```suggestion
The root caller is A and that is why it doesn't have a parent-id, only have a new trace-id. Next, A calls B using HTTP. To propagate the correlation information as part of the request, we are using two new headers based on the W3C Correlation specification, trace-id and parent-id. In this example because A is the root caller, A only sends its own trace-id to microservice B.
```